library-core/src/main/java/com/mikepenz/fastadapter/adapters/ItemFilter.kt

```diff
@@ -32,19 +32,10 @@ open class ItemFilter<Model, Item : GenericItem>(private val mItemAdapter: Model
      */
     open val selections: Set<Int>
         get() {
-            return originalItems?.let { originalItems ->
-                val selections = HashSet<Int>()
-                mItemAdapter.fastAdapter?.getPreItemCountByOrder(mItemAdapter.order)
-                        ?.let { adapterOffset ->
-                            originalItems.forEachIndexed { index, item ->
-                                if (item.isSelected) {
-                                    selections.add(index + adapterOffset)
-                                }
-                            }
-                        }
-                return selections
-            }
-                    ?: mItemAdapter.fastAdapter?.getExtension<SelectExtension<Item>>(SelectExtension::class.java)?.selections
+            val fastAdapter = mItemAdapter.fastAdapter ?: return emptySet()
+            val adapterOffset = fastAdapter.getPreItemCountByOrder(mItemAdapter.order)
+            return originalItems?.mapIndexedNotNull { index, item -> if (item.isSelected) index + adapterOffset else null }?.toSet()
```

Comment:
same as comment below

Reply:
Is this the reason for all of the collections? What about using sequences to map values, or using `mapTo`/`filterTo` for one liners? I think the function chains make it more readable, even if it's at the cost of sequences

Reply:
It's only a reason when chaining multiple operations. I don't think smaller code is a reason for reducing performance.

Reply:
The cost only comes when there are multiple operations. Otherwise  you just specify the initial collector. In this case I'll specify it to map using a `HashSet` and avoid calling `toSet()`

Reply:
54fd11c1

Reply:
Yeah filterTo is a good alternative to the previous .filter {}.toSet()